stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketUtil.java

```diff
@@ -224,6 +257,161 @@ public static int findIdleTimeoutInSeconds(MapValue<String, Object> annAttrIdleT
+                WebSocketConstants.CLIENT_CUSTOM_HEADERS_CONFIG);
+        if (headerValues != null) {
+            clientConnectorConfig.addHeaders(getCustomHeaders(headerValues));
+        }
+
+        long idleTimeoutInSeconds = WebSocketUtil.findIdleTimeoutInSeconds(clientEndpointConfig);
+        if (idleTimeoutInSeconds > 0) {
+            clientConnectorConfig.setIdleTimeoutInMillis((int) (idleTimeoutInSeconds * 1000));
+        }
+
+        clientConnectorConfig.setMaxFrameSize(WebSocketUtil.findMaxFrameSize(clientEndpointConfig));
+
+        MapValue secureSocket = clientEndpointConfig.getMapValue(HttpConstants.ENDPOINT_CONFIG_SECURE_SOCKET);
+        if (secureSocket != null) {
+            HttpUtil.populateSSLConfiguration(clientConnectorConfig, secureSocket);
+        } else {
+            HttpUtil.setDefaultTrustStore(clientConnectorConfig);
+        }
+    }
+
+    private static void establishReconnection(WebSocketClientConnector clientConnector, ObjectValue webSocketClient,
+                                              int maxReconnectInterval, WebSocketService wsService) {
+        CountDownLatch countDownLatch = (CountDownLatch) webSocketClient.getNativeData(COUNT_DOWN_LATCH);
+        waitForTime(countDownLatch, maxReconnectInterval, TimeUnit.MILLISECONDS);
+        establishWebSocketConnection(clientConnector, webSocketClient, wsService);
+        countDownLatch.countDown();
+    }
+
+    public static boolean reconnect(WebSocketOpenConnectionInfo connectionInfo) {
+        ObjectValue clientEndpointConfig = connectionInfo.getWebSocketEndpoint();
+        MapValue<String, Object> reconnectConfig = clientEndpointConfig.getMapValue(CLIENT_ENDPOINT_CONFIG).
+                getMapValue(RETRY_CONFIG);
+        WebSocketClientConnector clientConnector = (WebSocketClientConnector) clientEndpointConfig
+                .getNativeData(CLIENT_CONNECTOR);
+        int reconnectInterval = getIntegerValue(reconnectConfig.getIntValue(RECONNECT_INTERVAL));
+        Double reconnectDecay = reconnectConfig.getFloatValue(RETRY_DECAY);
+        int maxReconnectInterval = getIntegerValue(reconnectConfig.getIntValue(MAX_RETRY_INTERVAL));
+        int maxReconnectAttempts = getIntegerValue(reconnectConfig.getIntValue(MAX_RETRY_COUNT));
+        int reconnectAttempts = getIntegerValue(Long.valueOf(clientEndpointConfig.getNativeData(RECONNECT_ATTEMPTS).
+                toString()));
+        int timeout = (int) (reconnectInterval * Math.pow(reconnectDecay, reconnectAttempts));
+        if (timeout > maxReconnectInterval) {
+            maxReconnectInterval = timeout;
+        }
+        boolean attemptReconnect = false;
+        if (reconnectAttempts < maxReconnectAttempts) {
+            console.println(RECONNECTING);
+            attemptReconnect = true;
+            clientEndpointConfig.addNativeData(RECONNECT_ATTEMPTS, reconnectAttempts + 1);
+            establishReconnection(clientConnector, clientEndpointConfig, maxReconnectInterval,
```

Comment:
Normally the called method follows the caller. Shall we follow that convention and have the called methods after the caller?